### PR TITLE
Makes depicted place and category items unselectable for nearby place

### LINKED ISSUE #5317: When uploading via Nearby, make the item's depiction (and category) undeselectable
Steps to reproduce:

1. Go to Nearby
2. Select any Wikidata item on the map
3. Start adding a picture for it
4. At the depiction step, the Wikidata item is automatically selected
5. Problem: This Wikidata item can be deselected

If the Wikidata item has a category property (for instance https://www.wikidata.org/wiki/Q243 links to https://commons.wikimedia.org/wiki/Category:Eiffel_Tower), then the same problem exists: The category can be deselected.

Since a picture taken in Nearby must depict the Wikidata item, the app must prevent the item and category from being deselected by the user.

Unimportant note: Some may argue that if they select https://commons.wikimedia.org/wiki/Category:Eiffel_Tower_in_European_Union_livery_at_night then they do not need to select https://commons.wikimedia.org/wiki/Category:Eiffel_Tower , but this is very rare for an item to have Commons subcategories while having no P18, and de-parenting bots can take care of such cases. The benefits of making the category undeselectable are thus higher than the rare and small benefits.

### PR DESCRIPTION
**Pull request by srishti-R**
**Description (required)**

Fixes #5317 

What changes did you make and why?
Makes the place and category unselectable

**Tests performed (required)**

Tested {prodDebug} on {pixel 5 emulator} with API level {34, 30}.

**Screenshots (for UI changes only)**
greyed_check

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/repository/UploadRepository.java [hunk 1]
```diff
@@ -293,6 +293,23 @@ public Single<List<DepictedItem>> getPlaceDepictions() {
         return depictModel.getPlaceDepictions(new ArrayList<>(qids));
     }
 
+    /**
+     * Gets the category for each unique {@link Place} associated with an {@link UploadItem}
+     * from {@link #getUploads()}
+     *
+     * @return a single that provides the categories
+     */
+    public Single<List<CategoryItem>> getCategoryDepictions() {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/repository/UploadRepository.java:302]: How about renaming this to `getPlaceCategories`, would it make sense?

File: app/src/main/java/fr/free/nrw/commons/upload/categories/CategoriesPresenter.kt [hunk 2]
```diff
@@ -228,4 +232,34 @@ class CategoriesPresenter @Inject constructor(
...
+        categoryList.value?.toMutableList()
+            ?.let { toSelect + it }
+            ?.distinctBy(CategoryItem::name)
+            ?.let { categoryList.value = it }
+    }
+
+    override fun getCategories(): LiveData<List<CategoryItem>> {
+        return categoryList
+    }
+
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/categories/CategoriesPresenter.kt:257]: Would you mind adding Javadoc here?

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]
I just tested with a Wikidata item that has no associated category, it seems like a strange "Hidden" category is shown instead of nothing being pre-selected:

https://github.com/commons-app/apps-android-commons/assets/99590/927896ca-d8f8-4ab5-804b-01e71486b407

**Comment by srishti-R** [comment]
> I just tested with a Wikidata item that has no associated category (https://www.wikidata.org/wiki/Q105799610 embassy of Brazil in Costa Rica), it seems like a strange "Hidden" category is shown instead of nothing being pre-selected:
> 
> It seems to work great with Fukusho-ji (which has an associated category).

I thought this was intended, will remove.

**Comment by srishti-R** [comment]
@nicolas-raoul made the requested changes, "hidden" is no longer a category that appears checked when no category is there. Need some help with 2 failing tests inside CategoriesPresenterTests.kt, I have never written tests for RxJava before and I have used up every brain cell available to fix those two.

**Comment by nicolas-raoul** [comment]
Note: Just tested, it works great now, thanks! I did not actually upload any file but I will try to, probably in the next few days.

About the unit tests: Would you mind creating a new issue about it? Please include Git repo/branch info to make it easily reproducible. Thanks! :-)

**Comment by srishti-R** [comment]
@nicolas-raoul I was able to fix all the CategoriesPresenter failing tests as here #5327. One confirmation I needed to make this deselectable feature better. Suppose we upload a set of images that have exif attached, we confirm by saying "Yes" to "is this a photo of some place nearby?", after this, shouldn't the depiction and category also be unselectable? Currently its not. We can choose to select/unselect the depiction and category in this specific case.

**Review by nicolas-raoul — APPROVED** [review]
Tested again, it still works great (I did not actually finish the upload though as I did not have suitable pictures)

**Comment by nicolas-raoul** [comment]
When using Nearby, the app does assign the uploaded picture to the Wikidata item. Even if the user changes their mind they have no way to prevent this assignment from taking place. So if they change their mind, they MUST just cancel their upload.

On the other side, even after selecting "Yes" for "is this a photo of some place nearby?", the user can still modify everything (captions/descriptions/depictions/categories), and it will be like if they has selected "No". The Wikidata item is not modified.

Summary: Nearby upload MUST be about that Wikidata item. Normal upload does not have such restrictions.

**Pull request merged**